src/lambda.rs

```diff
@@ -301,7 +301,7 @@ where
     // for future work). To work around this, we capture any errors during this phase
     // and pass them into the handler function itself, so that it can raise the error
     // when the function is invoked.
-    let init_result = (|| async {
+    let init_result = async {
```

Comment:
**comment:** This is much nicer! Do you know if this syntax has always worked? I seem to recall wrestling with this code when I first wrote it to get it to behave how I wanted. I wonder if I just missed this simplification, or if something in the language has changed 🤔

Reply:
As far as I know the syntax `let example_future = async { ... }`; has been valid since Rust 1.39.0, released in November 2019. If you were working with Rust before this version, it's possible you might have missed this simplification at the time. Earlier versions required using a feature flag for async/await, so the language's capabilities have evolved significantly since then.